internal/powerplatform/services/licensing/api_licensing.go

```diff
@@ -170,4 +190,47 @@ func (client *LicensingClient) RemoveEnvironmentsToBillingPolicy(ctx context.Con
 	_, err := client.Api.Execute(ctx, "POST", apiUrl.String(), nil, environments, []int{http.StatusOK}, nil)
 
 	return err
-}
\ No newline at end of file
+}
+
+func (client *LicensingClient) DoWaitForLifecycleOperationStatus(ctx context.Context, billingPolicyDto *BillingPolicyDto) (*BillingPolicyDto, error) {
+	// Get the ID of the billing policy
+	billingId := billingPolicyDto.Id
+
+	// Define how long to wait between retries
+	retryAfter := time.Duration(5) * time.Second
+
+	// Define the maximum time to wait for the billing policy to reach a terminal state
+	timeout := time.Duration(10) * time.Minute
+
+	// Get the start time
+	startTime := time.Now()
+
+	// Loop until the billing policy status is Enabled or Disabled or the timeout is reached
+	for {
+		// Get the billing policy
+		billingPolicy, err := client.GetBillingPolicy(ctx, billingId)
+
+		// If there was an error, return it
+		if err != nil {
+			return nil, err
+		}
+
+		// If the billing policy status is Enabled or Disabled, return it
+		if billingPolicy.Status == "Enabled" || billingPolicy.Status == "Disabled" {
+			return billingPolicy, nil
+		}
+
+		// Check if the timeout is reached
+		if time.Since(startTime) >= timeout {
+			tflog.Debug(ctx, "Timeout reached while waiting for billing policy to reach a terminal state (Enabled or Disabled)")
+			err := fmt.Errorf("timeout reached while waiting for billing policy to reach a terminal state (Enabled or Disabled)")
+			return nil, err
+		}
+
+		// Wait before trying again
+		time.Sleep(retryAfter)
+
+		// Log that we are retrying
+		tflog.Debug(ctx, "Billing Policy Operation State: '"+billingPolicy.Status+"'")
```

Comment:
Use fmt.Sprintf(..) instead of string concatenation

Reply:
Code updated with recommendation 😊